src/semver.toit

```diff
@@ -111,3 +116,43 @@ compare a/string b/string [--if_equal]:
   comp := compare_dotted_ a_prerelease b_prerelease
   if comp != 0: return comp
   return if_equal.call
+
+/**
+Compares the semver strings $a and $b and returns true if $a < $b.
+
+This is a convenience function for `(compare a b) < 0`.
+*/
+is_less_than a/string b/string -> bool:
+  return (compare a b) < 0
+
+/**
+Compares the semver strings $a and $b and returns true if $a <= $b.
+
+This is a convenience function for `(compare a b) <= 0`.
+*/
+is_less_than_or_equal a/string b/string -> bool:
+  return (compare a b) <= 0
+
+/**
+Compares the semver strings $a and $b and returns true if $a == $b.
+
+This is a convenience function for `(compare a b) == 0`.
```

Comment:
When does `compare` return zero for different strings? Is it only for versions that lack certain parts (or have different v/V prefixes)? This might be a good place to document this.

Reply:
Primarily for the build suffix. `v1.0.0+abc` orders the same way as `v1.0.0+def`.

Probably an indication that we shouldn't use `equals` for the function.